File: php/src/Model/WorkerTeam.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/** One team membership as it appears on a worker's detail record. */
final class WorkerTeam
{
    public function __construct(
        public readonly string $teamId,
        public readonly string $key,
        public readonly string $name,
        public readonly ?string $color = null,
        /** 'member' | 'lead' */
        public readonly string $role = 'member',
    ) {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            (string) ($data['teamId'] ?? ''),
            (string) ($data['key'] ?? ''),
            (string) ($data['name'] ?? ''),
            isset($data['color']) ? (string) $data['color'] : null,
            (string) ($data['role'] ?? 'member'),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerTeam.java
package io.fivexer.sdk.model;

/** One team membership as it appears on a worker's detail record. */
public class WorkerTeam {

    private String teamId;
    private String key;
    private String name;
    private String color;
    private String role;

    public String getTeamId() { return teamId; }
    public String getKey() { return key; }
    public String getName() { return name; }
    public String getColor() { return color; }

    /** {@code "member"} or {@code "lead"}. */
    public String getRole() { return role; }
}
